### DISCUSSION TITLE
Guardar Datos CFDI

### DISCUSSION THREAD

**Discussion by asanderjs** [AndresRamos/Contpaqi#35] [Q&A] [answered]
Buenas

Tengo problemas para cambiar el Regimen Fiscal, Uso de CFDI, y Metodo de Pago de una Factura/Documento,
La documentacion del SDK no contienen una funcion para guardar esos datos en la Factura.
Se de donde obtenerlos, mas no como guardarlos.

Cualquier ayuda para apuntar en la direccion correcta seria muy apreciada. Gracias.

**Comment by AndresRamos** [comment, marked as answer]
Hola @asanderjs

A que te refieres cuando dices:

> Se de donde obtenerlos, mas no como guardarlos.

Asumiendo que estas usando las funciones `fLeeDato*` para leer los valores de esos campos, para asignar los datos deberías usar las funciones `fSetDato*` de los catálogos correspondientes.

Los datos **Regimen Fiscal** y **Uso de CFDI** se asignan en la tabla de clientes.
El **Metodo De Pago** se asigna en la tabla de documentos.

**Reply by asanderjs** [reply]
Buenas, gracias por responder

A lo que me referia a no saber como guardarlos es si es necesario modificar el Regimen Fiscal, y el Uso de CFDI en el cliente cada vez que se hace un nuevo Documento? o un cliente siempre va a llevar el mismo Regimen Fiscal y Uso de CFDI?

**Reply by AndresRamos** [reply]
Correcto, antes de crear un documento debes asignar al cliente su Regimen Fiscal y Uso de CFDI. Al momento de crear un documento el SDK asignara al documento los valores previamente asignados al cliente.

> un cliente siempre va a llevar el mismo Regimen Fiscal y Uso de CFDI?

El Regimen Fiscal se asigna con el SAT y no cambia. El Uso de CFDI si puede cambiar, pero por lo general utilizan el mismo. 

> es necesario modificar el Regimen Fiscal, y el Uso de CFDI en el cliente cada vez que se hace un nuevo Documento?

No. Si asignaste los valores correctos al cliente ya no es necesario volver a asignarlos. A menos que haya que hacer correcciones. Aqui lo importante es entender que los valores de esos datos deben estar asignados antes de generar el documento. Una vez generado el documento hay datos del CFDI que no se pueden modificar con el SDK (_*En Comercial si se pueden editar._) como lo son el Regimen Fiscal y el Uso de CFDI. El método de pago si se puede modificar. La recomendación es eliminar el documento y volverlo a crear correctamente.
